### ISSUE TITLE
Do we need the `Inputs` namespaces?

### ISSUE THREAD

**Issue by MisinformedDNA** [pulumi/pulumi#5501]
The `Inputs` namespaces are confusing and cumbersome. Take the below classes as an example:

```csharp
Pulumi.AzureNextGen.Web.V20150801.Site;
Pulumi.AzureNextGen.Web.V20150801.SiteArgs;
Pulumi.AzureNextGen.Web.V20150801.Inputs.SiteConfigArgs;
```

`Site` and `SiteArgs` are not in the `Inputs` namespace, but `SiteConfigArgs` is. 

So if I type:

```csharp
using Pulumi.AzureNextGen.Web.V20150801;

new Site("", new SiteArgs
{
    SiteConfig = new SiteConfigArgs();
});
```

I get an error message because `SiteConfigArgs` is in another namespace, which is not intuitive.

Why is it like this? Is it when an "args" object is only ever referenced in another "args" object?

As AzureNextGen is in preview, this may be a great time to reevaluate namespaces and, potentailly, begin deprecating the `Inputs` namespaces.

Thoughts?

**Comment by mikhailshilkov** [comment]
> Is it when an "args" object is only ever referenced in another "args" object?

This is correct. `Inputs` namespace contains types that are used in properties of resource arguments.

In TF-based providers, there are clashes of type names if they were to locate in the same namespace. However, this may not be relevant to Azure NextGen. We should look at what the best long-term solution may be.

**Comment by mikhailshilkov** [comment]
Found an example of a conflict in Azure NextGen:

- https://github.com/pulumi/pulumi-azure-nextgen/blob/master/sdk/dotnet/Network/Latest/NetworkSecurityGroup.cs#L168
- https://github.com/pulumi/pulumi-azure-nextgen/blob/master/sdk/dotnet/Network/Latest/Inputs/NetworkSecurityGroupArgs.cs#L16

**Comment by MisinformedDNA** [comment]
Hmm... makes sense. I filed https://github.com/pulumi/pulumi-azure-nextgen/issues/47 as a potential step in this direction.
